### PR TITLE
My First PR

### PR DESCRIPTION
**Pull request by FatemaAlshaikh**

### CODE REVIEW

File: CancelBooking UseCase/CancelBookinng.java [hunk 1]
```diff
@@ -4,6 +4,8 @@
 import java.time.LocalDate;
 // Data Layer
 // Represents a reservation with details like customer name, room type, room number, price, CPR, phone number, and booking date.
+
+
```
Comment (Fatema-Mohd) [inline CancelBooking UseCase/CancelBookinng.java]:
```suggestion
-you can make the getter methods more secure by making the field as "final". This ensures that once a -reservation is created, it can’t be modified. 
-you have to ensure input values are validated before storing them.
-(toStoring) method may reveal mush sensitive information like cpr, you can create more secure version or remove some sensitive details.
```

File: CancelBooking UseCase/CancelBookinng.java [hunk 2]
```diff
@@ -61,6 +63,8 @@ public String toString() {
     }
 }
 // Manages a collection of reservations, allowing adding, removing, and checking existence of reservations.
+
+
```
Comment (Fatema-Mohd) [inline CancelBooking UseCase/CancelBookinng.java]:
```suggestion
- the current code uses an "ArrayList", which is not thread-safe. in multi-threaded environment consider using "CopyOnWriteArrayList" or a synchronised block for critical method.
- you can use "HashMap" for fast lookups of reservations by room number and CPR combination.
```

File: CancelBooking UseCase/CancelBookinng.java [hunk 3]
```diff
@@ -86,6 +90,8 @@ public boolean exists(String roomNumber, String cpr) {
 }
 // Business Layer
 // Provides methods for reservation management, including adding, canceling, and displaying reservations.
+
+
```
Comment (Fatema-Mohd) [inline CancelBooking UseCase/CancelBookinng.java]:
```suggestion
you can improve the "processRefund" method to handle cases like no-show or cancellations after check-in.
```
```suggestion
instead of having a method handles both logic and user input/output (like cancelReservation), separate out logic from user-facing code
```

File: CancelBooking UseCase/CancelBookinng.java [hunk 4]
```diff
@@ -146,7 +152,9 @@ public void showCancellationPolicy() {
 }
 // Presentation Layer
 // Main class to handle user interaction for making and canceling reservations
-public class CancelBooking {
+
+
```
Comment (Fatema-Mohd) [inline CancelBooking UseCase/CancelBookinng.java]:
```suggestion
instead of dealing with logic and UI together in the "main" method, you can separate concerns into distinct methods and keep "main" only for managing control flow. also, you can add input validation for room number and CPR input to make sure they are not empty or invalid.
```

File: CancelBooking UseCase/ReservationTest.java [hunk 5]
```diff
@@ -1,12 +1,16 @@
 import org.junit.Test;
 import java.time.LocalDate;
 import static org.junit.Assert.*;
+
```
Comment (Fatema-Mohd) [inline CancelBooking UseCase/ReservationTest.java:4]:
```suggestion
the "Reservation" class has getters like (getCustomerName()), but in the tests you are calling (getcustomerName()), you have to ensure consistent naming. also, you can add test case, like the scenario when invalid data is passed. to the cnstructor.

```

File: CancelBooking UseCase/ReservationRepositoryTest.java [hunk 6]
```diff
@@ -1,10 +1,10 @@
 import org.junit.Test;
-
 import java.time.LocalDate;
-
 import static org.junit.Assert.*;
 
+
```
Comment (Fatema-Mohd) [inline CancelBooking UseCase/ReservationRepositoryTest.java]:
```suggestion
you can add an edge case for (remove) "test removing a reservation that does not exist". and you can add a test for removing an item when the repository is empty.

```

File: CancelBooking UseCase/ReservationServiceTest.java [hunk 7]
```diff
@@ -1,12 +1,12 @@
 import org.junit.Test;
-
 import java.time.LocalDate;
 import java.util.List;
 import java.util.Scanner;
-
 import static org.junit.Assert.*;
 
+
```
Comment (Fatema-Mohd) [inline CancelBooking UseCase/ReservationServiceTest.java]:
```suggestion
you can consider adding more scenarios where the user input is invalid or canceled, or where the user provides an unexpected response. also you can add a test for invalid inputs.

```

File: CancelBooking UseCase/CancelBookingTest.java [hunk 8]
```diff
@@ -1,23 +1,34 @@
...
-        }
-    }
+import org.junit.Test;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.time.LocalDate;
+import static org.junit.Assert.*;
+
+
+
```
Comment (Fatema-Mohd) [inline CancelBooking UseCase/CancelBookingTest.java]:
```suggestion
you can improve the console output testing by using a custom (PrintStream) in the test to capture system output and test more specific content in the console output. you can also test scenarios like when there are no reservations.
```

**Review by Fatema-Mohd — CHANGES_REQUESTED** [review]
I hope you develop and improve the code.
Thank you.

**Review by Fatema-Mohd — CHANGES_REQUESTED** [review]
I hope you develop and improve the code.
Thank you.
